### PR TITLE
Factor lightbox out into hook/context (react-query refactor)

### PR DESCRIPTION
**Pull request by pfrazee**

### CODE REVIEW

File: src/state/models/ui/shell.ts [hunk 1]
```diff
@@ -56,32 +29,13 @@ export class ShellUiModel {
...
-    this.rootStore.emitNavigation()
-    this.isLightboxActive = true
-    this.activeLightbox = lightbox
-  }
-
-  closeLightbox() {
-    this.isLightboxActive = false
-    this.activeLightbox = null
-  }
+  closeAllActiveElements() {}
```
Comment (estrattonbailey) [inline src/state/models/ui/shell.ts:35]: I see you have this tracked in the project, sweet

File: src/view/com/util/post-embeds/index.tsx [hunk 2]
```diff
@@ -19,8 +19,8 @@ import {
 } from '@atproto/api'
 import {Link} from '../Link'
 import {ImageLayoutGrid} from '../images/ImageLayoutGrid'
-import {ImagesLightbox} from 'state/models/ui/shell'
-import {useStores} from 'state/index'
+import {} from 'state/models/ui/shell'
```
Comment (estrattonbailey) [inline src/view/com/util/post-embeds/index.tsx:22]:
```suggestion
```

File: src/state/lightbox.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  openLightbox: () => {},
+  closeLightbox: () => {},
+})
+
+/**
+ * @deprecated DO NOT USE THIS unless you have no other choice.
+ */
+export let unstable__openLightbox: (lightbox: Lightbox) => void = () => {
+  throw new Error(`LightboxContext is not initialized`)
+}
```
Comment (estrattonbailey) [inline src/state/lightbox.tsx:45]: Unused in this PR right? Do we need it?

**Review by estrattonbailey — APPROVED** [review]
LGTM, tested on web/ios

File: src/state/lightbox.tsx (hunk 3 above)
Comment (pfrazee) [reply src/state/lightbox.tsx:45]: I guess not, but ironically I do need the close variant

**Pull request merged**